CAIPs/caip-19.md

```diff
@@ -0,0 +1,117 @@
+---
+caip: 19
+title: Asset Type and Asset ID Specification
+author: Antoine Herzog (@antoineherzog), Pedro Gomes (@pedrouid), Joel Thorstensson (@oed)
+discussions-to: https://github.com/ChainAgnostic/CAIPs/pull/19
+status: Draft
+type: Standard
+created: 2020-06-23
+updated: 2020-06-23
+requires: 10
+---
+
+## Simple Summary
+
+CAIP-19 defines a way to identify a type of asset (e.g. Bitcoin, Ether, ATOM) and an asset ID (for non fungible token) in a human readably, developer friendly and transaction-friendly way.
+
+## Abstract
+
+Often you need to reference the asset type or the asset ID. For example when you want to do an atomic swap between a fungible asset and a non fungible asset.
+
+## Motivation
+
+Currently, each wallet or each exchange needs to create their own registry of type of assets and their associated metadata for example like [Trust-Wallet](https://github.com/trustwallet/assets/tree/master/blockchains) or [CoinMarketCap](https://coinmarketcap.com/). Providing a unique type of Asset and an Asset ID for each asset for developers can reduce the risk of confusion betweens different assets. 
+
+## Specification of Asset Type
+
+The Asset Type is a string designed to uniquely identify type of assets in a developer-friendly fashion.
+
+### Syntax
+
+The `asset_type` is a case-sensitive string in the form
+
+```
+asset_type:    chain_id + "/" + asset_namespace + ":" + asset_reference
+chain_id:          Blockchain ID Specification cf. CAIP2 
+asset_namespace:   [-a-z0-9]{3,16}
+asset_reference:   [-a-zA-Z0-9]{1,47}
+```
+
+## Specification of Asset ID
+
+The Asset ID is a string designed to uniquely identify a non fungible asset in a developer-friendly fashion.
```

Comment:
Do we really want to limit it to "non fungible"? Also if ERC-20 is an asset we want to cover - isn't ERC-20 fungible?

Reply:
We can define an Asset Type for non fungible token and also fungible token: for example, DAI or CRYPTOKITTIES
When it comes to Asset ID, I believe we can give an ID only to non fungible token because for fungible token, they are fungible token, so we cannot attribute a specific ID for each coin?